Mixpanel/Mixpanel.m

```diff
@@ -359,6 +359,17 @@ - (void)identify:(NSString *)distinctId
         if ([Mixpanel inBackground]) {
             [self archiveProperties];
         }
+        
+        // Clear cached variants
+        self.variants = nil;
```

Comment:
Because if i'm not, the variants cached data of the previous user applied on the current user.

I also have a sample app for simulate this issue, if you want, I can send it to you.
BTW, I worked with Orien Wu from Mixpanel to solve this issue